File: CodeLocatorPlugin/src/main/java/com/bytedance/tools/codelocator/utils/ThreadUtils.java
package com.bytedance.tools.codelocator.utils;

import com.intellij.openapi.application.ApplicationManager;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class ThreadUtils {

    private static ExecutorService sExecutorService = Executors.newFixedThreadPool(3);

    private static ExecutorService sLogExecutorService = Executors.newSingleThreadExecutor();

    public static void submit(Runnable runnable) {
        sExecutorService.submit(runnable);
    }

    public static void submitLog(Runnable runnable) {
        sLogExecutorService.submit(runnable);
    }

    public static void runOnUIThread(Runnable runnable) {
        ApplicationManager.getApplication().invokeLater(runnable);
    }

}

File: CodeLocatorPlugin/src/main/java/com/bytedance/tools/codelocator/utils/ThreadUtils.kt
package com.bytedance.tools.codelocator.utils

import com.intellij.openapi.application.ApplicationManager
import java.util.concurrent.Executors

object ThreadUtils {

    private val sLogExecutorService = Executors.newSingleThreadExecutor()

    @JvmStatic
    fun submit(runnable: () -> Unit) {
        ApplicationManager.getApplication().executeOnPooledThread(runnable)
    }

    @JvmStatic
    fun submit(runnable: Runnable) {
        ApplicationManager.getApplication().executeOnPooledThread(runnable)
    }

    @JvmStatic
    fun submitLog(runnable: () -> Unit) {
        sLogExecutorService.submit(runnable)
    }

    @JvmStatic
    fun submitLog(runnable: Runnable) {
        sLogExecutorService.submit(runnable)
    }

    @JvmStatic
    fun runOnUIThread(runnable: () -> Unit) {
        ApplicationManager.getApplication().invokeLater(runnable)
    }

    @JvmStatic
    fun runOnUIThread(runnable: Runnable) {
        ApplicationManager.getApplication().invokeLater(runnable)
    }

}